### PR TITLE
rollback image can not be an image that errored

### PR DESCRIPTION
**Pull request by Saurabh-Thakre**
# Description

fixes [THEEDGE-2039] for now.

Fixes # (issue)

## Type of change

What is it?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [ ] Refactor

# Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I run `go fmt ./...` to check that my code is properly formatted
- [x] I run `go vet ./...` to check that my code is free of common Go style mistakes

### CODE REVIEW

File: pkg/services/images.go [hunk 1]
```diff
@@ -1197,7 +1197,7 @@ func (s *ImageService) GetRollbackImage(image *models.Image) (*models.Image, err
 		s.log.Error("Error retreving account")
 		return nil, new(AccountNotSet)
 	}
-	result := db.DB.Joins("Commit").Joins("Installer").Preload("Packages").Preload("CustomPackages").Preload("ThirdPartyRepositories").Preload("Commit.InstalledPackages").Preload("Commit.Repo").Where(&models.Image{ImageSetID: image.ImageSetID, Account: account}).Last(&rollback, "images.id < ?", image.ID)
+	result := db.DB.Joins("Commit").Joins("Installer").Preload("Packages").Preload("CustomPackages").Preload("ThirdPartyRepositories").Preload("Commit.InstalledPackages").Preload("Commit.Repo").Where(&models.Image{ImageSetID: image.ImageSetID, Account: account, Status: models.ImageStatusSuccess}).Last(&rollback, "images.id < ?", image.ID)
```
Comment (ldjebran) [inline pkg/services/images.go:1200]: is there a unitest that need to be changed and add the failing intermediate image, if does exists we need to add different cases .

Comment (Saurabh-Thakre) [reply pkg/services/images.go:1200]: yeah, lets refactor the current test to have another image v3 and imagev2 as ERROR

**Comment by InsightsDroid** [comment]
Test results are available in [Ibutsu](https://url.corp.redhat.com/ibutsu-runs). The test run IDs are:
21a4d131-b783-4de5-8380-c1ee88762f45
3afc5299-bf17-4bd5-8ce3-41c63d4285d6

**Review by juliasmf — APPROVED** [review]
LGTM

**Comment by de1987** [comment]
> LGTM

I was testing to check if we had vulnerability by using an external account to approve the PR and bypass mergify bot

**Comment by Saurabh-Thakre** [comment]
> > LGTM
> 
> I was testing to check if we had vulnerability by using an external account to approve the PR and bypass mergify bot

ohh heh nice one man, looks like the external account won't bypass mergify

**Review by ldjebran — APPROVED** [review]
LGMT

**Comment by InsightsDroid** [comment]
Test results are available in [Ibutsu](https://url.corp.redhat.com/ibutsu-runs). The test run IDs are:
81389354-a0df-4b7f-935d-174d6197a088
f0c225f1-40ff-4923-b25c-87ac12fb2354

**Pull request merged**